### PR TITLE
FUSETOOLS-2070 - improvements for error handling for deployment

### PR DESCRIPTION
**Pull request by lhein**

### CODE REVIEW

File: servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/server/subsystems/publish/ModuleBundleVersionUtility.java [hunk 1]
```diff
@@ -131,15 +140,18 @@ public boolean visit(IResource resource) throws CoreException {
...
-		File out = new File(path, file);
-		if (out.exists() && out.isFile()) {
-			try (JarFile jf = new JarFile(out)) {
-				bd = createBundleDetails(jf.getManifest());
-			} catch (IOException ex) {
-				Activator.getLogger().error(ex);
+		IMavenProjectFacade m2prjFacade = MavenPlugin.getMavenProjectRegistry().create(prj, new NullProgressMonitor());
+		MavenProject m2Project = m2prjFacade.getMavenProject();
+		if (m2Project != null) {
+			String path = m2prjFacade.getMavenProject().getBuild().getOutputDirectory();
```
Comment (apupier) [inline servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/server/subsystems/publish/ModuleBundleVersionUtility.java]: m2Project can be reused for m2prjFacade.getMavenProject()

File: servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/server/subsystems/Karaf2xPublishController.java [hunk 2]
```diff
@@ -211,6 +211,9 @@ public int transferBuiltModule(IModule[] module, IPath srcFile, IProgressMonitor
 			((Server)getServer()).setModulePublishState(module, IServer.PUBLISH_STATE_NONE);
 			((Server)getServer()).setServerPublishState(IServer.PUBLISH_STATE_NONE);			
 			status = IServer.PUBLISH_STATE_NONE;
+		} else {
+			Activator.getLogger().error("Deployment problem: Binary archive " + srcFile.toOSString() + " is not valid or lacking a valid MANIFEST.MF!");
```
Comment (apupier) [inline servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/server/subsystems/Karaf2xPublishController.java]: This sentence is the one provided in UI right?
Would be better to externalize it, if it is the case

**Comment by apupier** [comment]
done a quick review but didn't had time to test it locally. (implicit: everybody else can do it :-))

**Comment by lhein** [comment]
addressed your points. Final test is still open.

**Comment by apupier** [comment]
+1

**Pull request merged**